lib/jormungandr/src/Cardano/Pool/Jormungandr/Metrics.hs

```diff
@@ -163,7 +169,8 @@ data StakePoolLayer e m = StakePoolLayer
 -- The pool productions and stake distrubtions in the db can /never/ be from
 -- different forks such that it's safe for readers to access it.
 monitorStakePools
-    :: Tracer IO StakePoolLog
+    :: GetStakeDistribution t IO ~ GetStakeDistribution Jormungandr IO
```

Comment:
Any reason for not writing:

```hs
    :: GetStakeDistribution Jormungandr IO
```

?

Reply:
Yes - allowing other targets than `Jormungandr` in the tests :S  https://github.com/input-output-hk/cardano-wallet/blob/1fab8e511360af8e567a6f2fd4e9e7b77c3df917/lib/jormungandr/test/unit/Cardano/Pool/Jormungandr/MetricsSpec.hs#L278-L281

Reply:
I am thinking we should get rid of the `NetworkLayer` (or break it up) later and the need for the type families (and this weirdness)

Reply:
Okay, this is required for `module Cardano.Pool.Jormungandr.MetricsSpec` and the polymorphism on the target.

Reply:
> I am thinking we should get rid of the NetworkLayer (or break it up) later and the need for the type families (and this weirdness)

Although the source of the problem isn't much the `NetworkLayer` here, but the double-support Jormungandr / cardano-node :upside_down_face: 
Another abstraction could work better, but it'd still have to deal with backend discrepancies.